### PR TITLE
fix: remove all circular dependencies

### LINKED ISSUE #87: Circular dependency
I am using this library through `@azure/identity` and is throwing a `Circular dependency` warning. I have placed the text warning and picture.

I am importing `@azure` libraries as:
`import * as azureIdentity from '@azure/identity';`
`import * as appConfig from '@azure/app-configuration';`

Warning:
`Circular dependency: node_modules\@azure\identity\node_modules\@opentelemetry\api\build\src\index.js -> node_modules\@azure\inode_modules\@opentelemetry\api\build\src\internal\global-utils.js -> node_modules\@azure\identity\node_modules\@opentelemetr
Circular dependency: node_modules\@azure\identity\node_modules\@opentelemetry\api\build\src\index.js -> node_modules\@azure\inode_modules\@opentelemetry\api\build\src\internal\global-utils.js ->  C:\Users\JBlanco\Documents\MFE - Product\node_modules\\@azure\identity\node_modules\@opentelemetry\api\build\src\index.js
Circular dependency: node_modules\@azure\core-http\node_modules\@opentelemetry\api\build\src\index.js -> node_modules\@azure\tp\node_modules\@opentelemetry\api\build\src\internal\global-utils.js -> node_modules\@azure\core-http\node_modules\@opentele
Circular dependency: node_modules\@azure\core-http\node_modules\@opentelemetry\api\build\src\index.js -> node_modules\@azure\tp\node_modules\@opentelemetry\api\build\src\internal\global-utils.js ->  C:\Users\JBlanco\Documents\MFE - Product\node_modulules\@azure\core-http\node_modules\@opentelemetry\api\build\src\index.js`

image

Any other information needed please let me know.

### PR DESCRIPTION
**Pull request by dyladan**
Fixes #87

### CODE REVIEW

**Review by vmarchaud — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -63,6 +64,7 @@
     "@typescript-eslint/eslint-plugin": "4.28.1",
     "@typescript-eslint/parser": "4.28.1",
     "codecov": "3.8.2",
+    "dpdm": "^3.7.1",
```
Comment (Flarna) [inline package.json:67]:
```suggestion
    "dpdm": "3.7.1",
```

**Review by Flarna — APPROVED** [review]

File: src/trace/ProxyTracer.ts [hunk 2]
```diff
@@ -69,3 +68,7 @@ export class ProxyTracer implements Tracer {
     return this._delegate;
   }
 }
+
+export interface TracerDelegator {
```
[the remark this answers is not in the document]
Comment (dyladan) [reply src/trace/ProxyTracer.ts:72]: @obecny if you feel strong I can make the change, if not can you mark the conversation resolved so we can merge this?

**Review by obecny — APPROVED** [review]
lgtm, out of curiosity what does `dpdm` stands for ?, Suggestion maybe the job could be named to something more meaningful for example, `check:circular:dependency` :)

**Comment by dyladan** [comment]
> lgtm, out of curiosity what does `dpdm` stands for ?, Suggestion maybe the job could be named to something more meaningful for example, `check:circular:dependency` :)

No idea. The project aims to do more than just cycle checks. They also can look for missing dependencies and things like that (already checked by the typescript compiler in our case). I'll rename the check before merge.

**Pull request merged**